File: src/main/java/sh/talonfloof/dracoloader/api/Side.java
package sh.talonfloof.dracoloader.api;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.CLASS)
@Target({ElementType.TYPE, ElementType.METHOD, ElementType.FIELD, ElementType.CONSTRUCTOR, ElementType.PACKAGE})
public @interface Side {
    EnvironmentType value();
}

File: src/main/kotlin/sh/talonfloof/dracoloader/api/Side.kt
package sh.talonfloof.dracoloader.api

@Retention(AnnotationRetention.BINARY)
@Target(
    AnnotationTarget.CLASS,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.FIELD,
    AnnotationTarget.CONSTRUCTOR,
    AnnotationTarget.FILE
)
annotation class Side(val value: EnvironmentType)
